sigmastate/src/main/scala/sigmastate/lang/syntax/Identifiers.scala

```diff
@@ -14,43 +15,49 @@ object Identifiers {
   val OpCharNotSlash = NamedFunction(x => isOpChar(x) && x != '/')
   val NotBackTick = NamedFunction(_ != '`')
 
-  val Operator: Parser[Unit] = P(
+  def Operator[_:P]: P[Unit] = P(
     !Keywords ~ (!("/*" | "//") ~ (CharsWhile(OpCharNotSlash) | "/")).rep(1)
   )
 
-  val VarId: Parser[Unit] = VarId0(true)
+  def VarId[_:P]: P[Unit] = VarId0(true)
 
-  def VarId0(dollar: Boolean): Parser[Unit] = P( !Keywords ~ Lower ~ IdRest(dollar) )
-  val PlainId: Parser[Unit] = P( !Keywords ~ Upper ~ IdRest(true) | VarId | Operator ~ (!OpChar | &("/*" | "//")) )
-  val PlainIdNoDollar: Parser[Unit] = P( !Keywords ~ Upper ~ IdRest(false) | VarId0(false) | Operator )
-  val BacktickId: Parser[Unit] = P( "`" ~ CharsWhile(NotBackTick) ~ "`" )
-  val Id: P0 = P( BacktickId | PlainId )
+  def VarId0[_:P](dollar: Boolean): P[Unit] = P( !Keywords ~ Lower ~ IdRest(dollar) )
 
-  def IdRest(allowDollar: Boolean): Parser[Unit] = {
+  def UppercaseId[_: P](dollar: Boolean) = P( !Keywords ~ Upper ~ IdRest(dollar) )
```

Comment:
ScalaDocs missed in the trait

Reply:
added ScalaDocs and cleaned the code here https://github.com/ScorexFoundation/sigmastate-interpreter/commit/f6abcaef0f61c5066112eabcb27fb0c2ff5e3243